$NetBSD: patch-ak,v 1.3 2011/11/15 13:04:46 is Exp $

- use standard includes
- 64-bit time_t fixes
- remove union wait
- replace mktemp() by mkstemp()

--- src/in.fingerd.c.orig	1996-01-10 19:26:24.000000000 +0000
+++ src/in.fingerd.c
@@ -33,9 +33,7 @@ static char *rcsid = "$Id: in.fingerd.c,
 #include <sys/types.h>
 #include <fcntl.h>
 #include <sys/file.h>
-#ifndef X_OK
 #include <unistd.h>
-#endif
 #include <sys/stat.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
@@ -805,12 +803,12 @@ finger_users (user, options, stream)
 	  fprintf (stream, "%s (%s) seen at %s on %s",
 		   upackets[i]->real_name, upackets[i]->name,
 		   strip_hostname (upackets[i]->host),
-		   ctime (&upackets[i]->idle_time));
+		   ctime32 (&upackets[i]->idle_time));
 	else
 	  fprintf (stream, "%s seen at %s on %s",
 		   upackets[i]->name,
 		   strip_hostname (upackets[i]->host),
-		   ctime (&upackets[i]->idle_time));
+		   ctime32 (&upackets[i]->idle_time));
     }
   else
     {
@@ -1262,7 +1260,7 @@ show_unlogged_packet (packet, stream)
 
   /* In SunOS4 getpwnam() seems to call ctime(), so we move this call down
    * here to be sure the intended result doesn't get overwritten. */
-  char *the_time = ctime(&packet->idle_time);
+  char *the_time = ctime32(&packet->idle_time);
   the_time[strlen(the_time) - 1] = '\0';	/* delete newline */
 
   if (*(packet->real_name))
@@ -1406,14 +1404,7 @@ run_target_script(script, target, cd, in
   char *arg;
 {
   int pid;
-#ifdef SYSV
   int status;
-#else
-#ifndef WEXITSTATUS
-# define WEXITSTATUS(w) ((w).w_retcode)
-#endif
-  union wait status;
-#endif
 
   if (access (script, X_OK) >= 0)
     {
@@ -1528,6 +1519,7 @@ maybe_user_script (entry, stream, packet
   FILE *long_output;
   int result;
   extern char *mktemp();
+  int temp_fd;
 
 #ifndef FINGERRC
   return (0);
@@ -1571,8 +1563,9 @@ maybe_user_script (entry, stream, packet
   /* Collect regular long finger output in file */
   strcpy (temp_file, TEMPFILE);
   
-  if (!mktemp (temp_file) ||
-      !(long_output = fopen (temp_file, "w+")))
+  temp_fd = mkstemp (temp_file);
+  if (temp_fd < 0 ||
+      !(long_output = fdopen (temp_fd, "w+")))
     {
       file_error (WARNING, temp_file);
       free (user_script);
